internal/controller/grafanaorganization_controller.go

```diff
@@ -131,24 +131,64 @@ func (r GrafanaOrganizationReconciler) reconcileCreate(ctx context.Context, graf
 	}
 
 	// Update CR status if anything was changed
-	if organization.ID != grafanaOrganization.Status.OrgID {
-		grafanaOrganization.Status.OrgID = organization.ID
-
-		if err = r.Status().Update(ctx, grafanaOrganization); err != nil {
-			logger.Error(err, "failed to update the status")
-			return ctrl.Result{}, errors.WithStack(err)
-		}
+	if err = r.updateOrgStatus(ctx, grafanaOrganization, organization.ID); err != nil {
+		return ctrl.Result{}, errors.WithStack(err)
 	}
 
-	// TODO add datasources for the organization.
-
 	err = r.configureGrafana(ctx)
 	if err != nil {
 		return ctrl.Result{}, errors.WithStack(err)
 	}
 	return ctrl.Result{}, nil
 }
 
+func (r GrafanaOrganizationReconciler) updateOrgStatus(ctx context.Context, grafanaOrganization *v1alpha1.GrafanaOrganization, orgID int64) error {
+	logger := log.FromContext(ctx)
+
+	// Update orgID in the CR's satus
+	if grafanaOrganization.Status.OrgID != orgID {
```

Comment:
Can you add the orgstatus back in the main loop and make this a function dedicated to datasource creations?

Reply:
Done but not sure whether this matches what you expected.